providers/azure/aci.go

```diff
@@ -193,6 +195,37 @@ func (p *ACIProvider) GetPod(namespace, name string) (*v1.Pod, error) {
 	return containerGroupToPod(cg)
 }
 
+// GetPodLogs returns the logs of a pod by name that is running inside ACI.
+func (p *ACIProvider) GetContainerLogs(namespace, podName, containerName string) (string, error) {
+	logContent := ""
+	cg, err := p.aciClient.GetContainerGroup(p.resourceGroup, fmt.Sprintf("%s-%s", namespace, podName))
+	if err != nil {
+		// Trap error for 404 and return gracefully
+		if strings.Contains(err.Error(), "ResourceNotFound") {
+			return logContent, nil
+		}
+		return logContent, err
+	}
+
+	if cg.Tags["NodeName"] != p.nodeName {
+		return logContent, nil
+	}
+	// get logs from cg
+	retry := 10
+	for i := 0; i < retry; i++ {
+		cLogs, err := p.aciClient.GetContainerLogs(p.resourceGroup, cg.Name, containerName, 10)
```

Comment:
The tail lines would be better to be an input parameter of the GetContainerLogs function. 
I assume it Is in the query string of the URI from ApiServer, correct?

Reply:
side question, what else parameters will be in the URI?

Reply:
> query string of the URI from ApiServer

yes

Reply:
Just [two](https://github.com/ritazh/virtual-kubelet/blob/bcc5a33098f9a51839372e124e47eb928861a383/providers/azure/client/aci/logs.go#L17-L19)

Example:
https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerInstance/containerGroups/{containerGroupName}/containers/{containerName}/logs?api-version=2017-10-01-preview&tail={tail}

Reply:
I mean parameters in the request query string from the ApiServer to kubelet. ACI only supports tail right now. I want to know the gap.

Reply:
ah ok. from what i can tell, these are there in ACI:
- `kubectl logs pod` :   `GET /containerLogs/{namespace}/{pod}/{container}`
- `kubectl logs pod --tails=2` :   `GET /containerLogs/{namespace}/{pod}/{container}?tailLines=2`

But the following options are not there in ACI
- `kubectl logs pod -f` :   `GET /containerLogs/{namespace}/{pod}/{container}?follow=true` 
- `kubectl logs pod --since=1m` :   `GET /containerLogs/{namespace}/{pod}/{container}?sinceSeconds=60`
- `kubectl logs pod -p=true` :   `GET /containerLogs/{namespace}/{pod}/{container}?previous=true`
- `kubectl logs pod --limit-bytes=1` :   `GET /containerLogs/{namespace}/{pod}/{container}?limitBytes=1`
- `kubectl logs pod --timestamps=true` :   `GET /containerLogs/{namespace}/{pod}/{container}?timestamps=true`